src/index.js

```diff
@@ -1,75 +1,112 @@
-export { mix, is, like }
+'use strict';
 
-function mix(...args) {
-	var superclass = !is(args[0], 'factory') && args.shift() || baseclass
-	var factory = (is(args[args.length-1], 'factory') && args.pop()) || derive
-	superclass = is(superclass, 'mixin') ? superclass.class : derive(superclass)
-	if (args.length) factory = (org => superclass => org(args.reduce((s,m) => m.mixin(s), superclass)))(factory)
-	function mixin(superclass) {
-		var result = is(superclass, mixin) ? superclass : factory(superclass) 
-		if (mixin.classes.indexOf(result) === -1) mixin.classes.push(result)
-		return result
-	}
-	Object.defineProperties(mixin, {
-		classes: {value:[], writable:false},
-		mixins: {value:args, writable:false},
-	})
-	var Class = mixin(superclass)
-	var constructor = Class.hasOwnProperty('constructor') ? Class.constructor.bind(Class) : (...args) => new Class(...args)
-	Object.getOwnPropertyNames(Class).forEach(k => Object.defineProperty(constructor, k, {value: Class[k]}))
-	return Object.defineProperties(constructor, {
-		mixin: {value:mixin, writable:false},
-		class: {value: Class, writable:false},
-		interface: {get:(x => () => x ? x : x = getInterface(Class.prototype))()},
-	})
-}
+const baseclass = class Object {};
+const derive = superclass => ({}[superclass.name || 'Object'] = class extends superclass {});
+
+const getPropertyNames = function getPropertyNames(proto) {
```

Comment:
You might not believe it, but I actually think the original 75 lines of code were more easy to read then the new 112. That's probably a personal thing but I've noticed that I'm most productive if code is close together so as much as possible of it will fit on one screen. Having to scroll is actually hurting my productivity, which is why I will write `if (..) <statement>` on a single line and do more stuff like that that breaks convention.